modules/dgi_actions_handle/dgi_actions_handle.module

```diff
@@ -0,0 +1,172 @@
+<?php
+
+/**
+ * @file
+ * Holds hooks for DGI Actions handle.
+ */
+
+use Drupal\Core\Entity\EntityTypeInterface;
+use Drupal\Core\Form\FormStateInterface;
+use Drupal\node\NodeInterface;
+
+/**
+ * Function for adding unique constraint.
+ *
+ * This function is used to add a unique and required constraint
+ * to a specified field in an entity. If the 'suffix_field' is
+ * set and exists in the fields array, it adds a 'UniqueField' constraint to it.
+ *
+ * @param array $fields
+ *   The fields array.
+ * @param \Drupal\Core\Entity\EntityTypeInterface $entity_type
+ *   The entity type.
+ */
+function _dgi_actions_handle_suffix_validation_add_constraint(array &$fields, EntityTypeInterface $entity_type): void {
+  try {
+    $suffix_field = _dgi_actions_handle_get_suffix_field();
+    $handle_field = _dgi_actions_handle_get_handle_field();
+
+    // If the suffix and handle field is not set or the
+    // fields do not exist, return.
+    if (empty($fields[$suffix_field]) && empty($fields[$handle_field])) {
+      return;
+    }
+
+    // Add UniqueField and required constraint.
+    $fields[$suffix_field]->addConstraint('UniqueField');
+    $fields[$suffix_field]->setRequired(TRUE);
+
+    $fields[$handle_field]->addConstraint('UniqueField');
+  }
+  catch (Exception $e) {
+    Drupal::logger('dgi_actions_handle')->error($e->getMessage());
+  }
+}
+
+/**
+ * Function to get the suffix field.
+ *
+ * @return string|null
+ *   The suffix field if it is set, NULL otherwise.
+ *
+ * @throws \Drupal\Component\Plugin\Exception\InvalidPluginDefinitionException
+ * @throws \Drupal\Component\Plugin\Exception\PluginNotFoundException
+ */
+function _dgi_actions_handle_get_suffix_field(): string|null {
+  /** @var \Drupal\dgi_actions\Entity\ServiceDataInterface $handle_service_data */
+  $handle_service_data = Drupal::entityTypeManager()->getStorage('dgiactions_servicedata')->load('handles');
```

Comment:
This looks to be a left over from testing, same in https://github.com/discoverygarden/dgi_actions/pull/43/files#diff-f2c090fb44f2150f1e7d5487f2e8877dd3e1a70e280316f308f0b05901f428dcR166.